common/src/main/java/org/opensearch/sql/common/utils/QueryContext.java

```diff
@@ -48,6 +52,24 @@ public static String getRequestId() {
     return id;
   }
 
+  /**
+   * Capture error message to aggregate diagnostics
+   * for both legacy and new SQL engines.
+   * Can be deleted once the
+   * legacy SQL engine is deprecated.
+   */
+  public static void setError(String error) {
+    ThreadContext.put(ERROR_KEY, error);
```

Comment:
Wondering do we need to clean this up? I assume it's `ThreadLocal` underlying. Any risk if it's not reset properly?

Reply:
Thank you, I added resetting error in 4412631d. New function called when plugin starts processing a query.

Reply:
Cool. It's much safer now. :) Thanks!

Reply:
Btw, there is another use in TransportPPLQueryAction. Could you check if we also need to reset there and elsewhere as well?

Reply:
Ouch! Right, I fixed it in 03b0bcbe.